client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -369,7 +375,22 @@ class CelebornShuffleReader[K, C](
     }
   }
 
-  private def handleFetchExceptions(
+  @VisibleForTesting
+  def checkAndReportFetchFailureForUpdateFileGroupFailure(
+      celebornShuffleId: Int,
+      ce: Throwable): Unit = {
+    if (ce.getCause != null &&
+      (ce.getCause.isInstanceOf[InterruptedException] || ce.getCause.isInstanceOf[
+        TimeoutException])) {
```

Comment:
QQ: Does the `TimeoutException` only happen in the speculation task?

Reply:
A timeout Exception might happen if there is something wrong with the driver or the executor.
If the executor has something wrong, it might encounter a timeout exception which we will want the task to retry itself.
If the driver has something wrong, throw fetch failure might not be able to save the situation.
So in the current implementation, we assume that all timeout exceptions should be treated as normal exceptions and let the task retry itself.

Reply:
the current target branch is barnch-0.5, should it be main?

Reply:
As the main branch has your PR, this problem does not exist in the main branch. So the PR won't be cherry-picked to the main branch.

Reply:
Seems https://github.com/apache/celeborn/pull/2921 can handle the `InterruptedException`, but can not cover the `TimeoutException`.

Reply:
Yes, maybe another PR for the main to cover the timeout exception.

Reply:
I think it should be fine to backport the whole patch to main branch.